lua/doom/core/config/startup.lua

```diff
@@ -0,0 +1,41 @@
+local startup = {}
+
+local system = require("doom.core.system")
+local log = require("doom.extras.logging")
+
+-- Check if running Neovim or Vim and fails if:
+-- 1. Running Vim instead of Neovim
+-- 2. Running Neovim 0.4 or below
+if vim.fn.has("nvim") == 1 then
+  if vim.fn.has("nvim-0.5") ~= 1 then
+    log.fatal("Doom Nvim requires Neovim 0.5.0, please update it")
+  end
+else
+  log.fatal("Doom Nvim does not have support for Vim, please use it with Neovim instead")
+end
```

Comment:
We are already checking this in `lua/doom/core/config/init.lua` IIRC, it actually makes more sense to me to move this code to this file as you did.

However, we would need to actually call this file at top before start loading Doom core files so it will raise the error at the correct time IMO. Up to you what to do tho, deleting this chunk from this file or deleting this from the file that I told you in order to avoid duplicated code and reordering how are Doom files being loaded 🤔

Reply:
For now, I think leaving it on `config/init.lua` is better, because we want to fail fast on unsupported systems and this file really needs to be loaded last. This is actually only here due to careless copy-and-paste, but maybe it should be in a more logical place, such as the root `init.lua`.  I will look into it later.